File: Find-All-Numbers-Disappeared-in-an-Array.java
class Solution 
{
    public List<Integer> findDisappearedNumbers(int[] nums)
     {
        int n = nums.length;
        for (int i = 0; i < n; i++) 
        {
            int idx = Math.abs(nums[i]) - 1;
            nums[idx] = -Math.abs(nums[idx]);
        }

        List<Integer> res = new ArrayList<>();
        for (int i = 0; i < n; i++) 
        {
            if (nums[i] > 0)
                res.add(i + 1);
        }
        return res;
    }
}

File: Find-All-Numbers-Disappeared-in-an-Array.js
var findDisappearedNumbers = function(nums) 
{
    let n = nums.length;
    for (let i = 0; i < n; i++) 
    {
        let idx = Math.abs(nums[i]) - 1;
        nums[idx] = -Math.abs(nums[idx]);
    }

    let res = [];
    for (let i = 0; i < n; i++) 
    {
        if (nums[i] > 0) 
        {
            res.push(i + 1);
        }
    }
    return res;
};
